src/features/receive/AddressBlock.tsx

```diff
@@ -111,19 +124,50 @@ export const AddressBlock = (props: IAddressBlockProps): JSX.Element => {
     fetchAndSetSinglePrivKey,
     viewKey,
     fetchAndSetSingleViewKey,
+    transactions,
   } = props
 
   const { address } = addressBalance
   const balance = addressBalance.balance || 0
 
+  const getLastAccessed = () => {
+    if (!transactions) {
+      return null
+    }
+    const transaction = transactions.filter(
+      t => t?.address === address || t?.inputAddresses?.indexOf(address) !== -1,
+    )[0]
```

Comment:
Please simplify this by breaking into several statements

Reply:
It has been updated